cpmpy/expressions/globalconstraints.py

```diff
@@ -312,6 +312,146 @@ def value(self):
         return arrval in tab
 
 
+class MDD(GlobalConstraint):
+    """The values of the variables in 'array' correspond to a path in the mdd formed by the transition in 'transitions'
+    Root node is the first one used as a start in the first transition
+    """
+
+    def __init__(self, array, transitions):
+        array = flatlist(array)
+        if not all(isinstance(x, Expression) for x in array):
+            raise TypeError("the first argument of an MDD constraint should only contain variables/expressions")
+        super().__init__("mdd", [array, transitions])
```

Comment:
Should the validity of the "transition" argument not still be checked. Otherwise an error could occur later on, one that is not informative to the used (e.g. an error in the upcoming for loop what the structure of "transitions" is not as expected.